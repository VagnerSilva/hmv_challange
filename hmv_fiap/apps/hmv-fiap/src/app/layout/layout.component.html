<hmv-header></hmv-header>

<ng-content select="main"></ng-content>
