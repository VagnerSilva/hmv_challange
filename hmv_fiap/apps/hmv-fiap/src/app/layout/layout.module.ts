import { CommonModule } from '@angular/common'
import { NgModule } from '@angular/core'
import { SharedModule } from '@shared/shared.module'
import { HeaderComponent } from './components/header/header.component'
import { LayoutComponent } from './layout.component'

@NgModule({
	declarations: [LayoutComponent, HeaderComponent],
	imports: [CommonModule, SharedModule],
	exports: [LayoutComponent],
})
export class LayoutModule {}
