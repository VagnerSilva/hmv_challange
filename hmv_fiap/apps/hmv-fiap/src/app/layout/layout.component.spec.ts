import { Component } from '@angular/core'
import { createHostFactory, SpectatorHost } from '@ngneat/spectator/jest'
import { LayoutComponent } from './layout.component'

const cases = {
	single: `<hmv-layout></hmv-layout>`,
	main: `
  <hmv-layout>
    <main>{{title}}</main>
  </hmv-layout>`,
	title: 'Custom HostComponent',
}

@Component({ template: '' })
class CustomHostComponent {
	title = cases.title
}

describe('LayoutComponent', () => {
	let host: SpectatorHost<LayoutComponent, CustomHostComponent>
	const createHost = createHostFactory({
		component: LayoutComponent,
		host: CustomHostComponent,
	})

	it('Snapshot successfully', () => {
		host = createHost(cases.single)
		expect(host.hostFixture).toMatchSnapshot()
	})

	it('should display the host component title', () => {
		host = createHost(cases.main)
		expect(host.query(`main`)).toHaveText(cases.title)
		expect(host.hostFixture).toMatchSnapshot()
	})
})
