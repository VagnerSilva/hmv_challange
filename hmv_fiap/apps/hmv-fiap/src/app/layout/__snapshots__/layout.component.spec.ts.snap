// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutComponent Snapshot successfully 1`] = `
<ng-component
  __ngContext__={[Function LRootView]}
  title={[Function String]}
>
  <hmv-layout>
    <hmv-header />
  </hmv-layout>
</ng-component>
`;

exports[`LayoutComponent should display the host component title 1`] = `
<ng-component
  __ngContext__={[Function LRootView]}
  title={[Function String]}
>
  <hmv-layout>
    <hmv-header />
    <main>
      Custom HostComponent
    </main>
  </hmv-layout>
</ng-component>
`;
